refiner-lib/src/refiner_inner.rs

```diff
@@ -934,6 +1013,42 @@ fn build_transaction(
     })
 }
 
+fn determine_ft_on_transfer_recipient(
+    execution_outcome: &ExecutionOutcomeWithReceipt,
+    args: &NEP141FtOnTransferArgs,
+    storage: &Storage,
+    near_block: &BlockView,
+    transaction_index: u32,
+) -> Address {
+    match get_token_mint_kind(&execution_outcome.execution_outcome.outcome.logs) {
```

Comment:
Nit: `get_token_mint_kind` ends up being called twice in the flow of processing ft_on_transfer because it is called here and in setting value + input. We could consider getting the mint kind once and passing it as an argument to this function.

Reply:
Agreed, fixed.